### PR TITLE
Add validation to node tags

### LINKED ISSUE #2241: Add validation to node tags
## Description
Currently we do not validate or restrict the format of tags that can be added to nodes. This is inconsistent with other user-defined values. As part of #2014 it was agreed that tags should be subject to the same constraints as node names (see: https://github.com/kedro-org/kedro/blob/main/kedro/pipeline/node.py#L102)

### PR DESCRIPTION
**Pull request by AhdraMeraliQB**
> **NOTE:** Kedro datasets are moving from `kedro.extras.datasets` to a separate `kedro-datasets` package in
> [`kedro-plugins` repository](https://github.com/kedro-org/kedro-plugins). Any changes to the dataset implementations
> should be done by opening a pull request in that repository.
## Description

Resolves #2241 
## Developer Certificate of Origin
We need all contributions to comply with the [Developer Certificate of Origin (DCO)](https://developercertificate.org/). All commits must be signed off by including a `Signed-off-by` line in the commit message. [See our wiki for guidance](https://github.com/kedro-org/kedro/wiki/Guidelines-for-contributing-developers/).

If your PR is blocked due to unsigned commits, then you must follow the instructions under "Rebase the branch" on the GitHub Checks page for your PR. This will retroactively add the sign-off to all unsigned commits and allow the DCO check to pass.

## Checklist

- [x] Read the [contributing](https://github.com/kedro-org/kedro/blob/main/CONTRIBUTING.md) guidelines
- [x] Signed off each commit with a [Developer Certificate of Origin (DCO)](https://docs.github.com/en/repositories/managing-your-repositorys-settings-and-features/managing-repository-settings/managing-the-commit-signoff-policy-for-your-repository)
- [x] Opened this PR as a 'Draft Pull Request' if it is work-in-progress
- [ ] Updated the documentation to reflect the code changes
- [x] Added a description of this change in the [`RELEASE.md`](https://github.com/kedro-org/kedro/blob/main/RELEASE.md) file
- [x] Added tests to cover my changes
- [ ] Checked if this change will affect Kedro-Viz, and if so, communicated that with the Viz team

### CODE REVIEW

**Review by ankatiyar — APPROVED** [review]
Thanks @AhdraMeraliQB 💯

File: RELEASE.md [hunk 1]
```diff
@@ -11,7 +11,8 @@
 
 ## Bug fixes and other changes
 * Added a new field `add-ons` to `pyproject.toml` when a project is created.
-* Reduced `spaceflights` data to minimise waiting times during tutorial execution.
+* Reduced `spaceflights` data to minimize waiting times during tutorial execution.
```
Comment (SajidAlamQB) [inline RELEASE.md:14]:
```suggestion
* Reduced `spaceflights` data to minimise waiting times during tutorial execution.
```

Comment (SajidAlamQB) [reply RELEASE.md:14]: nit

**Review by SajidAlamQB — APPROVED** [review]
LGTM, minor nit-pick but otherwise all good, thank you @AhdraMeraliQB.

File: RELEASE.md (hunk 1 above)
Comment (AhdraMeraliQB) [reply RELEASE.md]: Thanks for catching this!

**Pull request merged**